### DISCUSSION TITLE
Could not find the declaration file for certain module.

### DISCUSSION THREAD

**Discussion by wlcharlie** [vercel/swr#2308] [Q&A] [answered]
# Bug report

## Description / Observed Behavior

POV: using Typescript

I was importing `useSWRMutation` from `swr/mutation` and the error occured: 

(Besides `mutation`, `immutable` and `infinite` has the same problem)

<img width="955" alt="image" src="https://user-images.githubusercontent.com/77925413/208593877-9a4566b4-32d0-42df-b616-42edac174cc1.png">

and I dug a bit found out the directory(exports) in package.json might be given the wrong path.

<img width="846" alt="image" src="https://user-images.githubusercontent.com/77925413/208594613-eb5f09e5-e8c4-43c5-8331-9d6b0f9b288c.png">

## Expected Behavior

How did you expect SWR to behave here?

Well, hope the type will work :D

## Repro Steps / Code Example

SWR version: 2.0.0

simply just follow the tutorial and with this line of code:
```typescript
import useSWRMutation from 'swr/mutation';
```

## Additional Context

the current workaround: I copied the d.ts into local d.ts

**Comment by wlcharlie** [comment]
and here's my tsconfig.json

```json
{
  "compilerOptions": {
    "target": "esnext",
    "baseUrl": ".",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noImplicitAny": true,
    "noEmit": true,
    "incremental": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "nodenext",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve"
  },
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", "**/*.js"],
  "exclude": ["node_modules"]
}
```

**Comment by wlcharlie** [comment, marked as answer]
Hi, @koba04 
I found out the bug report was moved to here. 
I was wondering is it my code set up wrong or SWR really did given the wrong path ?

thanks if reply :D

**Reply by koba04** [reply]
@wlcharlie Thank yo! This should have been kept in issues. I've fixed this at https://github.com/vercel/swr/pull/2341

**Reply by wlcharlie** [reply]
Wonderful, glad to help!
